src/Xamarin.Android.Build.Tasks/Tasks/ResolveAndroidTooling.cs

```diff
@@ -272,24 +272,38 @@ bool ValidateApiLevels ()
 				(string.IsNullOrWhiteSpace (AndroidApiLevel) && string.IsNullOrWhiteSpace (TargetFrameworkVersion));
 
 			if (UseLatestAndroidPlatformSdk) {
-				AndroidApiLevel = GetMaxInstalledApiLevel ().ToString ();
-				SupportedApiLevel = GetMaxStableApiLevel ().ToString ();
-				int maxInstalled, maxSupported = 0;
-				if (int.TryParse (AndroidApiLevel, out maxInstalled) && int.TryParse (SupportedApiLevel, out maxSupported) && maxInstalled > maxSupported) {
+				int maxInstalled = GetMaxInstalledApiLevel ();
+				int maxSupported = GetMaxStableApiLevel ();
+				AndroidApiLevel = maxInstalled.ToString ();
+				SupportedApiLevel = maxSupported.ToString ();
+				if (maxInstalled > maxSupported) {
 					Log.LogDebugMessage ($"API Level {AndroidApiLevel} is greater than the maximum supported API level of {SupportedApiLevel}. " +
 						"Support for this API will be added in a future release.");
-					AndroidApiLevel = SupportedApiLevel;
 				}
 				if (!string.IsNullOrWhiteSpace (TargetFrameworkVersion)) {
 					var userSelected = MonoAndroidHelper.SupportedVersions.GetApiLevelFromFrameworkVersion (TargetFrameworkVersion);
 					// overwrite using user version only if it is 
 					// above the maxStableApi and a valid apiLevel.
 					if (userSelected != null && userSelected > maxSupported && userSelected <= maxInstalled) {
-						AndroidApiLevel = userSelected.ToString ();
-						SupportedApiLevel = userSelected.ToString ();
+						maxInstalled =
+							maxSupported = userSelected.Value;
+						AndroidApiLevel =
+							SupportedApiLevel = userSelected.ToString ();
+					}
+				}
+
+				for (int apiLevel = maxSupported; apiLevel >= MonoAndroidHelper.SupportedVersions.MinStableVersion.ApiLevel; apiLevel--) {
+					var id = MonoAndroidHelper.SupportedVersions.GetIdFromApiLevel (apiLevel);
+					var apiPlatformDir = Path.Combine (AndroidSdkPath, "platforms", "android-" + id);
+					if (Directory.Exists (apiPlatformDir)) {
+						var targetFramework = MonoAndroidHelper.SupportedVersions.GetFrameworkVersionFromId (id);
+						if (targetFramework != null && MonoAndroidHelper.SupportedVersions.FrameworkDirectories.Any (p => Directory.Exists (Path.Combine (p, targetFramework)))) {
```

Comment:
Instead of using `.FrameworkDirectories`, I'd suggest using `SupportedVersions.InstalledBindingVersions.Any(b => b.FrameworkVersion == targetFramework)`